Formula/chapel.rb

```diff
@@ -23,14 +22,14 @@ class Chapel < Formula
   depends_on "python@3.9"
```

Comment:
does it work with python@3.10?

Reply:
I had to downgraded this in #103218 since `chapel` scripts expects `python3` to be available on PATH (comment about this is just above within collapsed diff). If we just set to `python@3.10`, it will break Linux.

Reply:
Excellent question!  We got bumped back to Python 3.9 in https://github.com/Homebrew/homebrew-core/commit/7722e6209e08ab62ef5d8b3119478aa854458ea1 for reasons that I have to admit were not entirely clear to me (most likely due to my own ignorance).  @tzinsky, have you happened to experiment with setting it back by any chance?

Reply:
I had tested it with the previous formula and we are still failing due to the PATH comment from @cho-m.